File: algo/Fibonacci/fib.c
#include <stdio.h>

int fib(int n) {
  int i, f[n + 1];
  f[0] = 1;
  f[1] = 1;
  for (i = 2; i <= n; i++)
    f[i] = f[i - 1] + f[i - 2];
  return f[n];
}

int main() {
  printf("%d\n", fib(10));
  return 0;
}

File: algo/Fibonacci/fib.rs
//Fibonacci

fn main()
{
    println!("{}", fib(10));
    
}

fn fib(n: i32) -> u64 {
	if n < 0 {
		panic!("{} is negative! negative Numbers not Support", n);
	}
	match n {
		0     => panic!("zero is not support"),
		1 | 2 => 1,
		3     => 2,
		_     => fib(n - 1) + fib(n - 2)
	}
}
